clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -85,6 +89,35 @@ public ConsumerRecord(String topic,
                           int serializedValueSize,
                           K key,
                           V value) {
+        this(topic, partition, offset, timestamp, timestampType, checksum, serializedKeySize, serializedValueSize, key, value, new RecordHeaders());
```

Comment:
So that it is set so if someone in their consumer interceptors consumes and needs to modify headers.

We could do null and then have a if null create lazily on the headers() method